### ISSUE TITLE
ui:repeat with begin and end continuous increment

### ISSUE THREAD

**Issue by enjanjoe** [eclipse-ee4j/mojarra#5150]
**Description**

Submitting the following form causes the repeat component to run two more times every time.

```
<h:form>
	<ui:repeat begin="1" end="5" varStatus="row">
		#{row.index}
	</ui:repeat>
	<br/>
	<h:commandButton value="Test"/>
</h:form>

```

**Output on page load**
0 1 2 3 4

**Output on 1st submit**
0 1 2 3 4 5 6

**Output on 2nd submit**
0 1 2 3 4 5 6 7 8

and it goes on..
using the step attribute with value > 1 causes even more erratic behavior.

Tested with WildFly 18.0.0.Final and Glassfish 5.0.1

**Comment by melloware** [comment]
Here is a runnable reproducer: 
[mojarra-5150.zip](https://github.com/eclipse-ee4j/mojarra/files/9484494/mojarra-5150.zip)

`mvn clean jetty:run -Pmojarra23` and then navigate to http://localhost:8080/primefaces-test/test.xhtml

**Comment by BalusC** [comment]
Sorry, I created branch but forgot PR.

**Comment by BalusC** [comment]
Merged into 2.3 and upmerged into 3.0 / 4.0
